From: Mike Christie <mchristi@redhat.com>
Date: Thu, 15 Jul 2010 21:34:51 -0400
Subject: [scsi] be2iscsi: fix for 64k data length sge
Message-id: <1279229691-16630-1-git-send-email-mchristi@redhat.com>
Patchwork-id: 26914
O-Subject: [PATCH RHEL 5.6]: be2iscsi: Fix for 64K data
Bugzilla: 608801
RH-Acked-by: Rob Evers <revers@redhat.com>

From: Mike Christie <mchristi@redhat.com>

    This is for BZ 608801.

    From upstream:
    This patch fixes a wraparound issue when the length=65536 for an
    sge. Since it was defined as a short, the length becomes zero

    This patch was made and tested by Emulex who have been doing
    the QE on the drive for us. I did some basic IO tests and it
    fixed the issue for me too.

diff --git a/drivers/scsi/be2iscsi/be_main.c b/drivers/scsi/be2iscsi/be_main.c
index 445e318..6658bd4 100644
--- a/drivers/scsi/be2iscsi/be_main.c
+++ b/drivers/scsi/be2iscsi/be_main.c
@@ -1777,7 +1777,7 @@ hwi_write_sgl(struct iscsi_wrb *pwrb, struct scatterlist *sg,
 	      unsigned int num_sg, struct beiscsi_io_task *io_task)
 {
 	struct iscsi_sge *psgl;
-	unsigned short sg_len, index;
+	unsigned int sg_len, index;
 	unsigned int sge_len = 0;
 	unsigned long long addr;
 	struct scatterlist *l_sg;
